redux-storage/redux-storage.d.ts

```diff
@@ -0,0 +1,107 @@
+// Type definitions for redux-storage 4.0.1
+// Project: https://github.com/michaelcontento/redux-storage
+// Definitions by: Alexey Svetliakov <https://github.com/asvetliakov>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../redux/redux.d.ts" />
+
+declare module "redux-storage" {
+    import { Reducer, Store, Middleware } from "redux";
+
+    /**
+     * Action constants
+     */
+    export const LOAD: string;
+    export const SAVE: string;
+
+    /**
+     * Storage engine interface
+     */
+    export interface StorageEngine {
+        /**
+         * Load
+         */
+        load(): PromiseLike<any>;
+        /**
+         * Save
+         * @param state
+         */
+        save(state: any): PromiseLike<any>
+    }
+
+    export interface StateMerger {
+        (oldState: any, newState: any): any;
+    }
+
+    /**
+     * Storage reducer
+     * @param reducer
+     * @param merger
+     */
+    export function reducer<TState>(reducer: Reducer<TState>, merger?: StateMerger): Reducer<TState>;
+
+    /**
+     * Create storage middleware
+     * @param engine
+     * @param actionBlacklist
+     * @param actionWhitelist
+     */
+    export function createMiddleware(engine: StorageEngine, actionBlacklist?: string[], actionWhitelist?: string[]): Middleware;
+
+    /**
+     * Loader interface
+     */
+    interface Loader<TState> {
+        (store: Store<TState>): PromiseLike<any>
+    }
+
+    /**
+     * Create state loader
+     * @param engine
+     */
+    export function createLoader<TState>(engine: StorageEngine): Loader<TState>;
+
+}
+
+declare module "redux-storage-decorator-filter" {
+    import { StorageEngine } from "redux-storage";
+
+    /**
+
+     */
+    interface WhitelistArg {
```

Comment:
Also if the `WhitelistArg` type is not exported, how does API consumer use it to fill the default function parameter?